docs/tools.md

```diff
@@ -0,0 +1,101 @@
+# Tools
+
+## Introduction
+
+Tools in the context of an agent refer to additional functionalities or capabilities integrated with the agent to perform specific tasks beyond text processing.
+
+These tools extend the agent's abilities, allowing it to interact with external systems, access information, and execute actions.
+
+## Development
+
+### Writing a new tool
+
+To create a tool, the `Tool` or `DynamicTool` class must be implemented from [tools base](../src/tools/base.ts). When starting to write tools, it is recommended to implement the `Tool` class. The `DynamicTool` class is an extension of the `Tool` class that allows for dynamic forms of input. Refer to the list of [examples](#examples) for some simple examples or any of the built-in tools in order to guide the creation of new tools.
+
+Tools MUST do the following:
+
+- Implement the `Tool` class:
+
+  ```typescript
+  export class MyNewTool extends Tool<MyNewToolOutput, ToolOptions, ToolRunOptions> {
+    // tool implementation
+  }
+  ```
+
+- Be given a unique name:
+
+  ```typescript
+  name = "MyNewTool";
```

Comment:
Mention that the best practice is to set the tool's `name` property to the name of its class.

Next things to mention
- `ToolOptions` is optional (default is `BaseToolOptions`) - constructor parameters that are passed during tool creation
- `ToolRunOptions` is optional (default is `BaseToolRunOptions`) - optional parameters that are passed to the `run` method.